src/Parser/Core/Modules/Items/BFA/FirstMatesSpyglass.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+import { formatPercentage } from 'common/format';
+
+/**
+ * First Mate's Spyglass -
+ * Use: Increase your Critical Strike by 768 for 15 sec. (2 Min Cooldown)
+ */
+class FirstMatesSpyglass extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  
+  casts = 0;
+  timesDamageCrit = 0;
+  timesHealCrit = 0;
+    
+  on_initialized() {
+    this.active = this.combatants.selected.hasTrinket(
+      ITEMS.FIRST_MATES_SPYGLASS.id
+    );
+  }
+  
+  on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.SPYGLASS_SIGHT.id) {
+    return;
+    }
+    this.casts += 1;
+  }
+  
+  on_byPlayer_damage(event) {
+    if (!this.combatants.selected.hasBuff(SPELLS.SPYGLASS_SIGHT.id)) {
+      return;
+    }
+    if (event.hitType !== HIT_TYPES.CRIT) {
+      return;
+    }
+    this.timesDamageCrit += 1;
+  }
+  
+  on_byPlayer_heal(event) {
+    if (!this.combatants.selected.hasBuff(SPELLS.SPYGLASS_SIGHT.id)) {
+      return;
+    }      
+    if (event.hitType !== HIT_TYPES.CRIT) {
+      return;
+      }
+      this.timesHealCrit += 1;
+  }
+  
+  get totalBuffUptime() {
+      return this.combatants.selected.getBuffUptime(SPELLS.SPYGLASS_SIGHT.id) / this.owner.fightDuration;
+  }
+  
+  item() {
+    return {
+      item: ITEMS.FIRST_MATES_SPYGLASS,
+      result: (
+        <dfn data-tip={`You critically damaged ${this.timesDamageCrit} times with this buff up<br/>
```

Comment:
Did you have any purpose in mind for this information? I think users can't really do anything with this information and so it seems unnecessary to me. We already present a huge amount of data (in most specs), so we need to keep it focused on actionable and interesting info.

Reply:
I was thinking that seeing the number of critical hits that happened while using an item that buffed your crit would be an interesting thing to note, kind of showing an incentive to use the item as much as possible when you want crits. 

Maybe changing it to "x times out of y hits," or a straight up percentage, would make it better? I'm not sure. I was just looking to see if there was any extra information someone might want out of using the buff.

Reply:
Showing the average amount of critical strike hits as a percentage of total hits could be interesting. 👍 

The most information (and the goal for all items) would be to show actual dps/hps gain, but unfortunately for items that provide stat rating buffs that requires the StatValues module to be implemented for specs and only a few have that. The next best thing we have for a bunch of items is the average stat gain, such as for [Aggramar's Conviction](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/c19e5ffa3404213059cd17f86a23c3dfe501f0e0/src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/AggramarsConviction.js#L82) and [Sephuz's Secret](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/c19e5ffa3404213059cd17f86a23c3dfe501f0e0/src/Parser/Core/Modules/Items/Legion/Legendaries/SephuzsSecret.js#L34).